### PR TITLE
Widget Configurator: Add custom token list

### PR DESCRIPTION
**Pull request by fairlighteth**
- Adds the option to select from multiple token lists
- CoW Protocol token list is enabled by default
- Option to add a custom token list
- Adds tokenlist to config
- Misc. styling fixes

https://github.com/cowprotocol/cowswap/assets/31534717/a461cd76-627c-45cd-a9b6-033b78917f87

## Need help
- Make the token lists actually work with the widget. Right now it simply adds the token lists to the configuration and does nothing else.
- Right now when adding a custom tokenlist it simply validates for a correct URL. Would be more robust to actually check if it is a usable token lists with detected tokens.

### CODE REVIEW

File: apps/widget-configurator/src/app/configurator/consts.ts [hunk 1]
```diff
@@ -1,3 +1,57 @@
...
+// Sourced from https://tokenlists.org/
+export const TOKEN_LISTS: TokenList[] = [
+  { name: 'CoW Protocol', url: 'https://files.cow.fi/tokens/CowSwap.json' },
+  { name: '1inch', url: 'https://tokens.1inch.eth.link' },
+  { name: 'Aave Token List', url: 'https://tokenlist.aave.eth.link' },
+  { name: 'Agora dataFi Tokens', url: 'https://datafi.theagora.eth.link' },
+  {
+    name: 'BA ERC20 SEC Action',
+    url: 'https://raw.githubusercontent.com/The-Blockchain-Association/sec-notice-list/master/ba-sec-list.json',
+  },
```
Comment (alfetopito) [inline apps/widget-configurator/src/app/configurator/consts.ts:14]: This list probably shouldn't be here

File: apps/widget-configurator/src/app/configurator/controls/TokenListControl.tsx [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+      '^(https?:\\/\\/)?' + // protocol
+        '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' + // domain name
+        '((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
+        '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
+        '(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
+        '(\\#[-a-z\\d_]*)?$',
+      'i'
+    ) // fragment locator
+    return !!pattern.test(url)
+  }
```
Comment (alfetopito) [inline apps/widget-configurator/src/app/configurator/controls/TokenListControl.tsx:70]: Is the logic in this file duplicated from somewhere?
I feel like we already have those validations somewhere

File: apps/widget-configurator/src/app/configurator/consts.ts (hunk 1 above)
Comment (anxolin) [reply apps/widget-configurator/src/app/configurator/consts.ts:14]: Why not?

**Review by anxolin — APPROVED** [review]
Soft approve! looks great but we would need to reiterate:
- To make the widget use this (we can help you with this), i will track it for when we get some capacity
- The state management has some issues. You are not supposed to mutate a global const. We should use state instead (we could help you with this)

**Comment by anxolin** [comment]
@fairlighteth do you think we could add some help message on how to make your own list? 
image

It's basically following the instructions in https://tokenlists.org, we can even provide one simple example to make it easier to start your list

File: apps/widget-configurator/src/app/configurator/consts.ts (hunk 1 above)
Comment (alfetopito) [reply apps/widget-configurator/src/app/configurator/consts.ts:14]: Do we want to have Uniswap's token block list as a list of tokens integrators want to enable in their UI?

**Review by shoom3301 — APPROVED** [review]

File: apps/widget-configurator/src/app/configurator/controls/TokenListControl.tsx [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+      setSelectedTokenLists(value)
+      onTokenListSelect(value)
+    },
+    [onTokenListSelect]
+  )
+
+  const handleAddCustomList = useCallback(
+    (newList: CustomList) => {
+      if (!TOKEN_LISTS.some((list) => list.url === newList.url)) {
+        TOKEN_LISTS.push(newList)
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply apps/widget-configurator/src/app/configurator/controls/TokenListControl.tsx]: Fixed

File: apps/widget-configurator/src/app/configurator/controls/TokenListControl.tsx (hunk 2 above)
Comment (shoom3301) [reply apps/widget-configurator/src/app/configurator/controls/TokenListControl.tsx:64]: Fixed

File: apps/widget-configurator/src/app/configurator/consts.ts (hunk 1 above)
Comment (shoom3301) [reply apps/widget-configurator/src/app/configurator/consts.ts]: Fixed

**Pull request merged**